app/src/main/java/com/adc/gpai/home/ForecasterScreen.kt

```diff
@@ -78,287 +92,729 @@ fun ForecasterScreen() {
+                    val locale = LocalContext.current.resources.configuration.locales.get(0)
+                    Text(
+                        text = "Cumulative GPA: ${String.format(locale, "%.2f", tempTranscript.gpa)}",
+                        fontSize = 24.sp,
+                        fontWeight = FontWeight.Bold,
+                        color = Color.White
+                    )
+                }
+            }
+
+            // Scrollable list of terms
+            LazyColumn(
+                modifier = Modifier.weight(1f),
+                verticalArrangement = Arrangement.spacedBy(12.dp)
+            ) {
+                itemsIndexed(tempTranscript.terms) { termIndex, term ->
+                    EnhancedTermSection(
+                        term = term,
+                        termIndex = termIndex,
+                        isCurrentSemester = termIndex == tempTranscript.terms.size - 1, // Last semester is current
+                        onUpdateTerm = { newTerm ->
+                            tempTranscript = tempTranscript.copy(
+                                terms = tempTranscript.terms.let {
+                                    val terms = it.toMutableList()
+                                    terms[termIndex] = newTerm
+                                    terms
+                                }
+                            )
+                        },
+                        onEditCourse = { courseIndex, course ->
+                            editingCourse = CourseEditState(
+                                termIndex = termIndex,
+                                courseIndex = courseIndex,
+                                course = course
+                            )
+                        },
+                        onDeleteCourse = { course ->
+                            courseToDelete = CourseDeleteState(course = course)
+                        },
+                        onAddCourse = {
+                            addingToTermIndex = termIndex
+                            showAddCourseDialog = true
+                        }
+                    )
+                }
+            }
         }
 
-        Column(modifier = Modifier.padding(16.dp)) {
-            val locale = LocalContext.current.resources.configuration.locales.get(0)
-            Text(
-                text = "Cumulative GPA: ${
-                    String.format(
-                        locale,
-                        "%.2f",
-                        tempTranscript.gpa
+        // Edit Course Dialog
+        if (editingCourse != null) {
+            CourseDialog(
+                title = "Edit Course",
+                initialCourseName = editingCourse!!.course.courseName,
+                initialCourseCode = editingCourse!!.course.courseCode,
+                initialCreditHours = editingCourse!!.course.attempted.toString(),
+                initialGrade = editingCourse!!.course.grade,
```

Comment:
seems like it would be a good idea to just pass in the entire course instead of having separate parameters for all these fields